util/validator.go

```diff
@@ -0,0 +1,45 @@
+package util
+
+import (
+	"regexp"
+	"strings"
+)
+
+var nameRegex = regexp.MustCompile(`^[a-z0-9](?:[a-z0-9-_@.]{0,61}[a-z0-9])?$`)
+
+// IsValidName
+// Compatible with Auth0's restricted character set (subset only: lowercase letters, numbers, hyphens, underscores, at (@), and dots (.))
+// Regex explanation:
+// ^                 // Start of string
+// [a-z0-9]          // First character must be a lowercase letter or digit
+// (?:               // Start non-capturing group:
+//
+//	[a-z0-9-]{1,61}  // Allow 1 to 61 lowercase letters, digits, or hyphens
+//	[a-z0-9]         // Final character must be a lowercase letter or digit
+//
+// )?                // Group is optional to allow 1-character usernames
+// $                 // End of string
+//
+// Notes:
+// - Prevents leading/trailing hyphens
+// - Does not allow special characters beyond a-z, 0-9, hyphens, underscores, at (@), and dots (.)
+// - Safe for use in subdomains and URL path segments
+// - Does NOT allow consecutive hyphens; add extra logic in Go if needed
+func IsValidName(name string) bool {
```

Comment:
@zombozo12 
Since it’s not necessarily limited to ‘name’, how about renaming it to IsSafePathName?